pkg/controllers/ippool/ippool_controller.go

```diff
@@ -134,10 +137,46 @@ func (r *IPPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 			log.V(1).Info("added finalizer on ippool CR", "ippool", req.NamespacedName)
 		}
 
+		if obj.Spec.Type == "" {
+			vpcNetworkConfig := common.ServiceMediator.GetVPCNetworkConfigByNamespace(obj.Namespace)
+			if vpcNetworkConfig == nil {
+				err := fmt.Errorf("operate failed: cannot get configuration for IPPool CR")
+				log.Error(nil, "failed to find VPCNetworkConfig for IPPool CR", "ippool", req.NamespacedName, "namespace %s", obj.Namespace)
```

Comment:
why use log.Error but error is nil?

Reply:
This is the way when we do not have error defined but want to print error message.
But I also saw Xie defined err in line 143, so I guess we can just leverage it?

Reply:
ok